doc/src/guides/solving/solve-ode.md

```diff
@@ -0,0 +1,423 @@
+derivative. Thus, all of the following `ypp` ("y prime prime") represent $y''$,
+the second derivative with respect to $x$ of a function $y(x)$:
+
+```py
+ypp = y(x).diff(x, x)
+ypp = y(x).diff(x, 2)
+ypp = y(x).diff((x, 2))
+ypp = diff(y(x), x, x)
+ypp = diff(y(x), x, 2)
+ypp = Derivative(y(x), x, x)
+ypp = Derivative(y(x), x, 2)
+ypp = Derivative(Derivative(y(y(x), x), x)
+ypp = diff(diff(y(x), x), x)
+yp = y(x).diff(x)
+ypp = yp.diff(x)
+```
+
+We recommend specifying the function to be solved for, as the second argument to
+{func}`~.dsolve`. Note that it must be a function rather than a variable
+(symbol). SymPy will give an error if you specify a variable ($x$) rather than a
+function ($f(x)$):
+
+```py
+>>> dsolve(Derivative(y(x), x, x) + 9*y(x), x)
+Traceback (most recent call last):
+    ...
+ValueError: dsolve() and classify_ode() only work with functions of one variable, not x
+```
+
+Similarly, you must specify the argument of the function: $f(x)$, not just $f$.
+
+You can define the function to be solved for in two ways. The subsequent syntax
+for specifying initial conditions depends on your choice.
+
+## Define a Function Without Including Its Independent Variable
+
+As in the example above, you can define a function without including its
+independent variable:
+
+```py
+>>> from sympy import symbols, Eq, Function, dsolve
+>>> f, g = symbols("f g", cls=Function)
+>>> x = symbols("x")
+>>> eqs = [Eq(f(x).diff(x), g(x)), Eq(g(x).diff(x), f(x))]
+>>> dsolve(eqs, [f(x), g(x)])
+[Eq(f(x), -C1*exp(-x) + C2*exp(x)), Eq(g(x), C1*exp(-x) + C2*exp(x))]
+```
+
+Note that you supply the functions to be solved for as a list as the second
+argument of {func}`~.dsolve`, here `[f(x), g(x)]`.
+
+### Specify Initial (Boundary) Conditions
+
+If your differential equation(s) have initial or boundary conditions, specify
+them with the {func}`~.dsolve` optional argument `ics`. It should be given in
+the form of `{f(x0): y0, f(x).diff(x).subs(x, x1): y1}` and so on where, for
+example, the value of `f(x)` at $x = $ `x0` is `y0`. For power series solutions,
+if no initial conditions are specified `f(0)` is assumed to be `C0` and the
+power series solution is calculated about $0$.
+
+Here is an example of setting the initial values for functions, namely namely
+$f(0) = 1$ and $g(2) = 3$:
+
+```py
+>>> from sympy import symbols, Eq, Function, dsolve
+>>> f, g = symbols("f g", cls=Function)
+>>> x = symbols("x")
+>>> eqs = [Eq(f(x).diff(x), g(x)), Eq(g(x).diff(x), f(x))]
+>>> dsolve(eqs, [f(x), g(x)])
+[Eq(f(x), -C1*exp(-x) + C2*exp(x)), Eq(g(x), C1*exp(-x) + C2*exp(x))]
+>>> dsolve(eqs, [f(x), g(x)], ics={f(0): 1, g(2): 3})
+[Eq(f(x), (1 + 3*exp(2))*exp(x)/(1 + exp(4)) - (-exp(4) + 3*exp(2))*exp(-x)/(1 + exp(4))), Eq(g(x), (1 + 3*exp(2))*exp(x)/(1 + exp(4)) + (-exp(4) + 3*exp(2))*exp(-x)/(1 + exp(4)))]
+```
+
+Here is an example of setting the initial value for the derivative of a
+function, namely $f'(1) = 2$:
```

Comment:
Even though it's supported, I don't think a typical use-case would be to underspecify the initial conditions like this. I would either also add a second condition or change this example to use a single ODE.

Reply:
Either specify an initial condition for g'(x), or just use one equation.